spec/requests/tags_request_spec.rb

```diff
@@ -0,0 +1,119 @@
+require 'rails_helper'
+require 'json'
+
+RSpec.describe 'Tag requests', type: :request do
+  let(:parsed_response)         { JSON.parse(response.body) }
+  let(:article)                 { create(:article) }
+  let(:tag)                     { create(:tag) }
+  let(:valid_tag_attributes)    { { title: 'valid' } }
+  let(:invalid_tag_attributes)  { { invalid_title: 'invalid' } }
+
+  describe 'GET /index' do
+    let(:first_article)           { create(:article, title: 'First article') }
+    let(:second_article)          { create(:article, title: 'Second article') }
+
+    before do
+      titles = %w[funny sad boring]
+      # Attach 3 tags to first article
+      titles.each do |title|
+        first_article.tags << create(:tag, title: title.to_s)
+      end
+      # Create and attach 'common' tag to both articles
+      first_article.tags << create(:tag, title: 'common')
+      second_article.tags << create(:tag, title: 'common')
+    end
+
+    context 'no query and no article_id is passed ' do
+      it 'returns http success and contains all created tags in json response' do
+        get tags_path
+
+        expect(response).to have_http_status(:ok)
+        expect(parsed_response.map do |tag|
+                 tag['title']
+               end).to match_array(%w[funny sad boring common common])
+      end
+    end
+
+    context 'query and no article_id is passed ' do
+      it 'returns http success and contains all created tags filtered by query in json response' do
+        get tags_path, params: { query: 'common' }
```

Comment:
```suggestion
        get tags_path, params: { query: 'com' }
```
To show partial matching works